apps/dashboard/src/components/primitives/form/form.tsx

```diff
@@ -108,31 +112,58 @@ const FormControl = React.forwardRef<React.ElementRef<typeof Slot>, React.Compon
 );
 FormControl.displayName = 'FormControl';
 
-const FormMessagePure = React.forwardRef<
-  HTMLParagraphElement,
-  React.HTMLAttributes<HTMLParagraphElement> & { error?: string }
->(({ className, children, error, id, ...props }, _ref) => {
-  const body = error ? error : children;
+type FormMessagePureProps = React.HTMLAttributes<HTMLParagraphElement> & { hasError?: boolean; icon?: IconType };
 
-  if (!body) {
-    return null;
+const FormMessagePure = React.forwardRef<HTMLParagraphElement, FormMessagePureProps>(
+  ({ className, children, hasError = false, icon, ...props }, _ref) => {
+    return (
+      children && (
+        <Hint hasError={hasError} {...props}>
+          {icon && <HintIcon as={icon} />}
+          {children}
+        </Hint>
+      )
+    );
   }
-
-  return (
-    <Hint hasError={!!error} {...props}>
-      <HintIcon as={error ? RiErrorWarningFill : RiInformationFill} />
-      {body}
-    </Hint>
-  );
-});
+);
 FormMessagePure.displayName = 'FormMessagePure';
 
-const FormMessage = React.forwardRef<HTMLParagraphElement, React.HTMLAttributes<HTMLParagraphElement>>((props, ref) => {
-  const { error, formMessageId } = useFormField();
+const FormMessage = React.forwardRef<HTMLParagraphElement, React.HTMLAttributes<HTMLParagraphElement>>(
+  ({ children, ...rest }, ref) => {
+    const { error, formMessageId } = useFormField();
+    const content = error ? String(error.message) : children;
+    const icon = error ? RiErrorWarningFill : RiInformationLine;
 
-  return <FormMessagePure ref={ref} id={formMessageId} error={error ? String(error?.message) : undefined} {...props} />;
-});
-FormMessage.displayName = 'FormMessage';
+    const isFirstMount = useRef(true);
```

Comment:
Animate Form messages (errors or hints). This animation ensures that transitioning from a hint to an error message or vice versa is always smooth.